### PR TITLE
Allow "disableAlpha" for color Picker component to be optional (replaces #5835)

### LINKED ISSUE #4726: ColorPalette: enable alpha channel
## Issue Overview
The Color Palette block restricts the alpha editing for colors. It will be very useful to make the attribute to enable/disable alpha while it is just the one parameter on React Color.

## Expected Behavior
The simple attribute passed on the block creation.
`<ColorPalette disableAlpha={ false } />`

## Current Behavior
At the moment, the alpha is restricted in ChromePicker creation in the block.
`<ChromePicker
   color={ value }
   onChangeComplete={ ( color ) => onChange( color.hex ) }
   style={ { width: '100%' } }
   **disableAlpha** />`

## Screenshots
Used to be.
image

### PR DESCRIPTION
**Pull request by diegoliv**
## Description
Currently, the `<ColorPalette />` component enforces the alpha channel from `react-color` to be disabled. Like described on #5811, there are benefits of giving developers the option to enable the alpha channel inside the color picker.

Previously #5835
Fixes #4726

## Tested With
WordPress 4.9.4
Gutenberg 2.4.0
PHP 7.0.3 / nginx
MySQL 5.5.55

## Types of changes
This PR introduces the ability to conditionally set a `disableAlpha` parameter for the `<ColorPallete />` component. It is set to `false` by default so it won't conflict with existing blocks that uses this component. Also, the returned color string depends on how the alpha channel inside the color picker is set. If opacity is 100%, we just return the `hex` value for the color. If opacity has a different level, then we return the `rgba()` value for the color.

### CODE REVIEW

**Comment by diegoliv** [comment]
@danielbachhuber here is an updated PR that replaces #5835. Take a look and let me know if is there anything to be tweaked.

**Review by danielbachhuber — APPROVED** [review]
👍 Looks good to me! Flagging a second review.

File: components/color-palette/index.js
Comment (tofumatt) [inline components/color-palette/index.js]: A nitpick but it might be nice to sort these props alphabetically. The current ordering makes it harder to sort through arguments and makes this last one feel especially tacked-on.

File: components/color-palette/index.js
Comment (tofumatt) [inline components/color-palette/index.js]: You could use template strings here and omit the join; it'd probably be a bit more readable:

```js
colorString = `rgba(${ r }, ${ g }, ${ b }, ${ a })`;
```

I'm not sure why the `Array.join` is being used at all here, it just seems to make it more complex.

File: components/color-palette/index.js
Comment (tofumatt) [inline components/color-palette/index.js]: Seems there's a code path here where `colorString` is `undefined`, should we test for its truthiness before running `onChange`?

**Review by tofumatt — COMMENTED** [review]
Largely seems fine but there are a few tweaks I think you could make to tidy up this code.

File: components/color-palette/index.js
Comment (danielbachhuber) [reply components/color-palette/index.js]: > Seems there's a code path here where `colorString` is `undefined`

What's the code path?

Comment (tofumatt) [reply components/color-palette/index.js]: Oh wait, there isn't, but because it's defined as `undefined` above it sort of read like maybe there could be.

I actually see why it's this way, never mind me, all good!

File: components/color-palette/index.js
Comment (diegoliv) [reply components/color-palette/index.js]: @tofumatt done!

File: components/color-palette/index.js
Comment (diegoliv) [reply components/color-palette/index.js]: @tofumatt yeah, I guess this is me still not 100% used to use ES6 features. But I totally agree with you. Just switched the variable to template strings.

**Comment by danielbachhuber** [comment]
Holding on merge per #7246. Sorry about that, @diegoliv — I appreciate the work you've put into this.

**Comment by gziolo** [comment]
Moving this to 3.2 Milestone as it seems to be blocked with #7246. Do I read it properly @danielbachhuber? Feel free to remove the label I'm about to add.

**Comment by danielbachhuber** [comment]
Yes, we're currently blocked by #7246

**Comment by TheoL** [comment]
#7246 has been fixed. Can you now proceed with this?

**Comment by tofumatt** [comment]
I'm fine with that but we probably won't look to merge it until after 5.0 ships as it's a reasonable UI change and we are trying to freeze the UI–at least as much as possible. 😄

**Comment by TheoL** [comment]
disableAlpha will still default to false so from a UI perspective it shouldn't be an issue - existing code will continue to work in the same way.

I need this for a plugin I am creating which I was hoping to release when 5.0 is officially out.

**Comment by gziolo** [comment]
It needs to be rebased with the latest changes from `master` branch.

**Comment by earnjam** [comment]
I just ran into this issue. It would be nice to tidy up the branch and try to get this shipped. @diegoliv are you still working on this?

**Review by gziolo — CHANGES_REQUESTED** [review]
@diegoliv, that would be great to get it up to date 👍

**Comment by gziolo** [comment]
This Pull Request wasn't updated for quite some time. Let's close it to let other contributors work on it. @diegoliv you can still refresh your work and ask for another round of review. It looks like this proposal is very close to be ready. Thank you for your contribution so far!

**Pull request closed without merging**
